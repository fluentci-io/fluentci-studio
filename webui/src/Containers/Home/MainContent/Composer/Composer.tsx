import { FC, useState } from "react";
import { PlusLg } from "@styled-icons/bootstrap";
import { Terminal } from "@styled-icons/bootstrap";
import { EllipsisVertical } from "@styled-icons/fa-solid";
import { StatefulPopover } from "baseui/popover";
import NewActionModal from "./NewActionModal";
import SetupActionModal from "./SetupActionModal";
import { Pipeline } from "./NewActionModal/NewActionModalWithData";
import {
  Action,
  ActionName,
  Connector,
  ConnectorContainer,
  PlusButton,
  PopoverButton,
} from "./styles";
import { StatefulMenu } from "baseui/menu";

export type ComposerProps = {
  actions: Pipeline[];
  setActions: (values: Pipeline[]) => void;
};

const Composer: FC<ComposerProps> = (props) => {
  const { actions, setActions } = props;
  const [isOpen, setIsOpen] = useState(false);
  const [selectedAction, setSelectedAction] = useState<Pipeline | null>(null);
  const [isSetupActionModalOpen, setIsSetupActionModalOpen] = useState(false);
  const [editAction, setEditAction] = useState(false);
  const [actionPosition, setActionPosition] = useState<number | null>(null);
  const [clickedPosition, setClickedPosition] = useState(0);

  function close() {
    setIsOpen(false);
  }

  function onAddNewAction(item: Pipeline) {
    setIsOpen(false);
    setIsSetupActionModalOpen(false);
    const updated = [...actions];
    updated.splice(clickedPosition, 0, item);
    setActions(updated);
  }

  function onClickAction(action: Pipeline, index: number) {
    setActionPosition(index);
    setSelectedAction(action);
    setEditAction(true);
    setIsSetupActionModalOpen(true);
  }

  function onDelete(position: number) {
    const updated = actions.filter((_, index) => index !== position);
    setActions(updated);
  }

  function onDuplicate(position: number) {
    const updated = [...actions];
    updated.splice(position, 0, actions[position]);
    setActions(updated);
  }

  return (
    <div style={{ display: "flex", flexDirection: "column", marginTop: 20 }}>
      {actions.length === 0 && (
        <PlusButton
          onClick={() => {
            setEditAction(false);
            setIsOpen(true);
            setClickedPosition(0);
          }}
        >
          <PlusLg size={15} color="#fff" />
        </PlusButton>
      )}
      {actions.map((action, index) => (
        <div key={action.id}>
          <PlusButton
            onClick={() => {
              setEditAction(false);
              setIsOpen(true);
              setClickedPosition(index);
            }}
          >
            <PlusLg size={15} color="#fff" />
          </PlusButton>
          <ConnectorContainer>
            <Connector />
          </ConnectorContainer>
          <Action
            onClick={(e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
              if (
                (e.target as HTMLElement).closest("svg") ||
                (e.target as HTMLElement).closest("button") ||
                (e.target as HTMLElement).closest("li")
              ) {
                return;
              }
              onClickAction(action, index ? 1 : index);
            }}
          >
            {action.logo && (
              <img
                src={action.logo}
                alt={action.name}
                style={{
                  backgroundColor:
                    action.name.startsWith("deno") ||
                    action.name.startsWith("github") ||
                    action.name.startsWith("rust") ||
                    action.name.startsWith("trivy") ||
                    action.name.startsWith("symfony") ||
                    action.name.startsWith("flakestry") ||
                    action.name.startsWith("symfony") ||
                    action.name.startsWith("heroku") ||
                    action.name.startsWith("django") ||
                    action.name.startsWith("terraform") ||
                    action.name.startsWith("prisma")
                      ? "#fff"
                      : "initial",
                  maxWidth: 34,
                  borderRadius: 2,
                  padding: 5,
                  marginRight: 15,
                }}
              />
            )}
            {!action.logo && (
              <Terminal
                size={32}
                color="#fff"
                style={{ marginLeft: 4, marginRight: 15 }}
              />
            )}
            <div style={{ display: "flex", flex: 1, overflow: "hidden" }}>
              <ActionName>
                {action.actionName || action.command?.split("\n")?.reverse()[0]}
              </ActionName>
            </div>
            <StatefulPopover
              placement="bottomRight"
              content={({ close }) => (
                <StatefulMenu
                  items={[
                    { label: <span style={{ color: "#fff" }}>Duplicate</span> },
                    {
                      label: (
                        <span style={{ color: "#ff0094" }}>Delete action</span>
                      ),
                    },
                  ]}
                  overrides={{
                    List: {
                      style: {
                        border: "1px solid rgb(71 5 94 / 29%)",
                      },
                    },
                    Option: {
                      props: {
                        overrides: {
                          ListItem: {
                            style: ({
                              $theme,
                            }: {
                              $theme: { primaryFontFamily: string };
                            }) => ({
                              fontFamily: $theme.primaryFontFamily,
                            }),
                          },
                        },
                      },
                    },
                  }}
                  onItemSelect={({
                    item,
                  }: {
                    item: { label: { props: { children: string } } };
                  }) => {
                    switch (item.label.props.children) {
                      case "Delete action":
                        onDelete(index);
                        break;
                      case "Duplicate":
                        onDuplicate(index);
                        break;
                      default:
                        break;
                    }
                    close();
                  }}
                />
              )}
              accessibilityType={"tooltip"}
            >
              <PopoverButton onClick={(e) => e.stopPropagation()}>
                <EllipsisVertical size={20} />
              </PopoverButton>
            </StatefulPopover>
          </Action>
          <ConnectorContainer>
            <Connector />
          </ConnectorContainer>
          {index === actions.length - 1 && (
            <PlusButton
              onClick={() => {
                setEditAction(false);
                setClickedPosition(index + 1);
                setIsOpen(true);
              }}
            >
              <PlusLg size={15} color="#fff" />
            </PlusButton>
          )}
        </div>
      ))}
      <NewActionModal
        onClose={close}
        isOpen={isOpen}
        onAdd={(item: Pipeline) => {
          setIsSetupActionModalOpen(true);
          setSelectedAction(item);
        }}
      />
      <SetupActionModal
        onClose={() => setIsSetupActionModalOpen(false)}
        isOpen={isSetupActionModalOpen}
        onAddThisAction={onAddNewAction}
        selectedAction={selectedAction!}
        editAction={editAction}
        actionPosition={actionPosition}
        onSaveChanges={() => {
          setIsSetupActionModalOpen(false);
        }}
      />
    </div>
  );
};

Composer.defaultProps = {
  actions: [],
};

export default Composer;
