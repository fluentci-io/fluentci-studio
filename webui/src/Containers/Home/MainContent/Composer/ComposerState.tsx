import { atom } from "recoil";
import { Pipeline } from "./NewActionModal/NewActionModalWithData";

export const ComposerState = atom<Pipeline[]>({
  key: "ComposerState",
  default: [],
});
