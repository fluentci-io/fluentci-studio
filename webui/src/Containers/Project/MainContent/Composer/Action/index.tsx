import Action from "./ActionWithData";

export default Action;
