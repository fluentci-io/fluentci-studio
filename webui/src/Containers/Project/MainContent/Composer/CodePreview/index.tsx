import CodePreview from "./CodePreviewWithData";

export default CodePreview;
