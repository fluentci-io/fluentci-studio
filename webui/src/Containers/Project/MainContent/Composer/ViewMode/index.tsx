import ViewMode from "./ViewModeWithData";

export default ViewMode;
