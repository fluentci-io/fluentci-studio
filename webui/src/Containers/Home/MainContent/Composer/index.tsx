import Composer from "./ComposerWithData";

export default Composer;
