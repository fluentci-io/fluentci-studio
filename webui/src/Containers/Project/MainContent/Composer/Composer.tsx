import { FC, useState } from "react";
import { PlusLg } from "@styled-icons/bootstrap";
import Action from "./Action";
import NewActionModal from "./NewActionModal";
import SetupActionModal from "./SetupActionModal";
import { Pipeline } from "./NewActionModal/NewActionModalWithData";
import { Connector, ConnectorContainer, PlusButton } from "./styles";
import ViewMode from "./ViewMode";
import CodePreview from "./CodePreview";

export type ComposerProps = {
  actions: Pipeline[];
  setActions: (values: Pipeline[]) => void;
  mode: "stacked" | "code";
};

const Composer: FC<ComposerProps> = (props) => {
  const { actions, setActions, mode } = props;
  const [isOpen, setIsOpen] = useState(false);
  const [selectedAction, setSelectedAction] = useState<Pipeline | null>(null);
  const [isSetupActionModalOpen, setIsSetupActionModalOpen] = useState(false);
  const [editAction, setEditAction] = useState(false);
  const [actionPosition, setActionPosition] = useState<number | null>(null);
  const [clickedPosition, setClickedPosition] = useState(0);

  function close() {
    setIsOpen(false);
  }

  function onAddNewAction(item: Pipeline) {
    setIsOpen(false);
    setIsSetupActionModalOpen(false);
    const updated = [...actions];
    updated.splice(clickedPosition, 0, item);
    setActions(updated);
  }

  function onClickAction(action: Pipeline, index: number) {
    setActionPosition(index);
    setSelectedAction(action);
    setEditAction(true);
    setIsSetupActionModalOpen(true);
  }

  function onDelete(position: number) {
    const updated = actions.filter((_, index) => index !== position);
    setActions(updated);
  }

  function onDuplicate(position: number) {
    const updated = [...actions];
    const newAction = actions[position];
    updated.splice(position, 0, newAction);
    setActions(updated);
  }

  return (
    <div
      style={{
        display: "flex",
        flexDirection: "column",
        position: "relative",
      }}
    >
      <ViewMode />
      {mode === "stacked" && (
        <div style={{ marginTop: 50 }}>
          {actions.length === 0 && (
            <PlusButton
              onClick={() => {
                setEditAction(false);
                setIsOpen(true);
                setClickedPosition(0);
              }}
            >
              <PlusLg size={15} color="#fff" />
            </PlusButton>
          )}
          {actions.map((action, index) => (
            <div key={index.toString()}>
              <PlusButton
                onClick={() => {
                  setEditAction(false);
                  setIsOpen(true);
                  setClickedPosition(index);
                }}
              >
                <PlusLg size={15} color="#fff" />
              </PlusButton>
              <ConnectorContainer>
                <Connector />
              </ConnectorContainer>
              <Action
                action={action}
                index={index}
                onClickAction={onClickAction}
                onDelete={onDelete}
                onDuplicate={onDuplicate}
              />
              <ConnectorContainer>
                <Connector />
              </ConnectorContainer>
              {index === actions.length - 1 && (
                <PlusButton
                  onClick={() => {
                    setEditAction(false);
                    setClickedPosition(index + 1);
                    setIsOpen(true);
                  }}
                >
                  <PlusLg size={15} color="#fff" />
                </PlusButton>
              )}
            </div>
          ))}
        </div>
      )}
      {mode === "code" && <CodePreview />}
      <NewActionModal
        onClose={close}
        isOpen={isOpen}
        onAdd={(item: Pipeline) => {
          setIsSetupActionModalOpen(true);
          setSelectedAction(item);
        }}
      />
      <SetupActionModal
        onClose={() => setIsSetupActionModalOpen(false)}
        isOpen={isSetupActionModalOpen}
        onAddThisAction={onAddNewAction}
        selectedAction={selectedAction!}
        editAction={editAction}
        actionPosition={actionPosition}
        onSaveChanges={() => {
          setIsSetupActionModalOpen(false);
        }}
      />
    </div>
  );
};

Composer.defaultProps = {
  actions: [],
};

export default Composer;
