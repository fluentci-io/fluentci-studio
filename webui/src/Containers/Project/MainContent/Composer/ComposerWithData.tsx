import { FC, useEffect } from "react";
import Composer from "./Composer";
import { useRecoilState, useRecoilValue } from "recoil";
import { ComposerState } from "./ComposerState";
import { Pipeline } from "./NewActionModal/NewActionModalWithData";
import {
  useGetActionsQuery,
  useSaveActionsMutation,
} from "../../../../Hooks/GraphQL";
import { useParams } from "react-router-dom";
import { ViewModeState } from "./ViewMode/ViewModeState";

const ComposerWithData: FC = () => {
  const { id } = useParams();
  const { data, refetch } = useGetActionsQuery({
    variables: {
      projectId: id!,
    },
  });
  const [saveActions] = useSaveActionsMutation();
  const mode = useRecoilValue(ViewModeState);
  const [actions, setActions] = useRecoilState(ComposerState);
  const _setActions = async (actions: Pipeline[]) => {
    setActions(actions);
    await saveActions({
      variables: {
        projectId: id!,
        actions: actions.map((action) => ({
          commands: action.command,
          enabled: !!action.active,
          logo: action.logo,
          name: action.actionName!,
          plugin: action.name!,
          useWasm: !!action.useWasmPlugin,
          githubUrl: action.githubUrl,
        })),
      },
    });
    await refetch();
  };

  useEffect(() => {
    if (data?.actions) {
      setActions(
        data.actions.map((action) => ({
          id: "",
          name: action.plugin,
          actionName: action.name,
          active: action.enabled,
          useWasmPlugin: action.useWasm,
          command: action.commands,
          description: "",
          githubUrl: action.githubUrl || "",
          logo: action.logo || "",
          packageId: action.plugin,
          downloads: 0,
          version: "",
          license: "",
          defaultBranch: "",
          updatedAt: "",
          comingSoon: false,
        }))
      );
      return;
    }
    setActions([]);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [data?.actions]);

  useEffect(() => {
    refetch();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return <Composer actions={actions} setActions={_setActions} mode={mode} />;
};

export default ComposerWithData;
