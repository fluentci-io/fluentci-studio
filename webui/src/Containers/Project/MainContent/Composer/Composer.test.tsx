import { render } from "@testing-library/react";
import Composer from "./Composer";
import { RecoilRoot } from "recoil";
import { MockedProvider } from "@apollo/client/testing";

describe("Composer", () => {
  it("should render", () => {
    const { container } = render(
      <Composer
        actions={[
          {
            id: "tyzlq1mhhx3haeay79m4owhf",
            name: "gleam",
            actionName: "test",
            active: true,
            useWasmPlugin: true,
            command: "test",
            description: "",
            githubUrl: "https://github.com/fluent-ci-templates/gleam-pipeline",
            logo: "https://cdn.jsdelivr.net/gh/fluent-ci-templates/.github/assets/gleam.png",
            packageId: "gleam",
            downloads: 0,
            version: "",
            license: "",
            defaultBranch: "",
            updatedAt: "",
            comingSoon: false,
          },
          {
            id: "qv9um6van8o271gyggzbmps5",
            name: "gleam",
            actionName: "build",
            active: true,
            useWasmPlugin: true,
            command: "build",
            description: "",
            githubUrl: "https://github.com/fluent-ci-templates/gleam-pipeline",
            logo: "https://cdn.jsdelivr.net/gh/fluent-ci-templates/.github/assets/gleam.png",
            packageId: "gleam",
            downloads: 0,
            version: "",
            license: "",
            defaultBranch: "",
            updatedAt: "",
            comingSoon: false,
          },
        ]}
        mode="stacked"
        setActions={vi.fn()}
      />,
      {
        wrapper: ({ children }) => (
          <MockedProvider mocks={[]}>
            <RecoilRoot>{children}</RecoilRoot>
          </MockedProvider>
        ),
      }
    );
    expect(container).toMatchSnapshot();
  });
});
