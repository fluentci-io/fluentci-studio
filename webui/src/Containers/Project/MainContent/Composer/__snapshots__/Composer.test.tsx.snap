// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Composer > should render 1`] = `
<div>
  <div
    style="display: flex; flex-direction: column; position: relative;"
  >
    <div
      class="css-1rwt18y"
    >
      <button
        class="css-olxsem"
      >
        <svg
          aria-hidden="true"
          class="StyledIconBase-sc-ea9ulj-0 gEpWWb"
          color="#fff"
          fill="currentColor"
          focusable="false"
          height="15"
          viewBox="0 0 16 16"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 0h10a2 2 0 0 1 2 2v3a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2V2a2 2 0 0 1 2-2zm0 1a1 1 0 0 0-1 1v3a1 1 0 0 0 1 1h10a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1H3zm0 8h10a2 2 0 0 1 2 2v3a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2v-3a2 2 0 0 1 2-2zm0 1a1 1 0 0 0-1 1v3a1 1 0 0 0 1 1h10a1 1 0 0 0 1-1v-3a1 1 0 0 0-1-1H3z"
          />
        </svg>
      </button>
      <button
        class="css-ualqix"
      >
        <svg
          aria-hidden="true"
          class="StyledIconBase-sc-ea9ulj-0 gEpWWb"
          color="#fff"
          fill="currentColor"
          focusable="false"
          height="21"
          viewBox="0 0 16 16"
          width="21"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.854 4.854a.5.5 0 1 0-.708-.708l-3.5 3.5a.5.5 0 0 0 0 .708l3.5 3.5a.5.5 0 0 0 .708-.708L2.707 8l3.147-3.146zm4.292 0a.5.5 0 0 1 .708-.708l3.5 3.5a.5.5 0 0 1 0 .708l-3.5 3.5a.5.5 0 0 1-.708-.708L13.293 8l-3.147-3.146z"
          />
        </svg>
      </button>
    </div>
    <div
      style="margin-top: 50px;"
    >
      <div>
        <button
          class="css-10yefha"
        >
          <svg
            aria-hidden="true"
            class="StyledIconBase-sc-ea9ulj-0 gEpWWb"
            color="#fff"
            fill="currentColor"
            focusable="false"
            height="15"
            viewBox="0 0 16 16"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 2a.5.5 0 0 1 .5.5v5h5a.5.5 0 0 1 0 1h-5v5a.5.5 0 0 1-1 0v-5h-5a.5.5 0 0 1 0-1h5v-5A.5.5 0 0 1 8 2Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <div
          class="css-12l3k25"
        >
          <div
            class="css-of0uoe"
          />
        </div>
        <div
          class="action css-1gebutb"
        >
          <img
            alt="gleam"
            class="action"
            src="https://cdn.jsdelivr.net/gh/fluent-ci-templates/.github/assets/gleam.png"
            style="max-width: 34px; border-radius: 2px; padding: 5px; margin-right: 15px;"
          />
          <div
            class="action"
            style="display: flex; flex: 1; overflow: hidden;"
          >
            <div
              class="css-3azhy3"
            >
              test
            </div>
          </div>
          <label
            class=""
            data-baseweb="checkbox"
          >
            <div
              class=""
            >
              <div
                class=""
              />
            </div>
            <input
              checked=""
              class=""
              type="checkbox"
              value=""
            />
          </label>
          <button
            class="css-1xnjjfw"
            id="1__anchor"
          >
            <svg
              aria-hidden="true"
              class="StyledIconBase-sc-ea9ulj-0 gEpWWb"
              fill="currentColor"
              focusable="false"
              height="20"
              viewBox="0 0 128 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M64 360c30.9 0 56 25.1 56 56s-25.1 56-56 56-56-25.1-56-56 25.1-56 56-56zm0-160c30.9 0 56 25.1 56 56s-25.1 56-56 56-56-25.1-56-56 25.1-56 56-56zm56-104c0 30.9-25.1 56-56 56S8 126.9 8 96s25.1-56 56-56 56 25.1 56 56z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="css-12l3k25"
        >
          <div
            class="css-of0uoe"
          />
        </div>
      </div>
      <div>
        <button
          class="css-10yefha"
        >
          <svg
            aria-hidden="true"
            class="StyledIconBase-sc-ea9ulj-0 gEpWWb"
            color="#fff"
            fill="currentColor"
            focusable="false"
            height="15"
            viewBox="0 0 16 16"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 2a.5.5 0 0 1 .5.5v5h5a.5.5 0 0 1 0 1h-5v5a.5.5 0 0 1-1 0v-5h-5a.5.5 0 0 1 0-1h5v-5A.5.5 0 0 1 8 2Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <div
          class="css-12l3k25"
        >
          <div
            class="css-of0uoe"
          />
        </div>
        <div
          class="action css-1gebutb"
        >
          <img
            alt="gleam"
            class="action"
            src="https://cdn.jsdelivr.net/gh/fluent-ci-templates/.github/assets/gleam.png"
            style="max-width: 34px; border-radius: 2px; padding: 5px; margin-right: 15px;"
          />
          <div
            class="action"
            style="display: flex; flex: 1; overflow: hidden;"
          >
            <div
              class="css-3azhy3"
            >
              build
            </div>
          </div>
          <label
            class=""
            data-baseweb="checkbox"
          >
            <div
              class=""
            >
              <div
                class=""
              />
            </div>
            <input
              checked=""
              class=""
              type="checkbox"
              value=""
            />
          </label>
          <button
            class="css-1xnjjfw"
            id="2__anchor"
          >
            <svg
              aria-hidden="true"
              class="StyledIconBase-sc-ea9ulj-0 gEpWWb"
              fill="currentColor"
              focusable="false"
              height="20"
              viewBox="0 0 128 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M64 360c30.9 0 56 25.1 56 56s-25.1 56-56 56-56-25.1-56-56 25.1-56 56-56zm0-160c30.9 0 56 25.1 56 56s-25.1 56-56 56-56-25.1-56-56 25.1-56 56-56zm56-104c0 30.9-25.1 56-56 56S8 126.9 8 96s25.1-56 56-56 56 25.1 56 56z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="css-12l3k25"
        >
          <div
            class="css-of0uoe"
          />
        </div>
        <button
          class="css-10yefha"
        >
          <svg
            aria-hidden="true"
            class="StyledIconBase-sc-ea9ulj-0 gEpWWb"
            color="#fff"
            fill="currentColor"
            focusable="false"
            height="15"
            viewBox="0 0 16 16"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 2a.5.5 0 0 1 .5.5v5h5a.5.5 0 0 1 0 1h-5v5a.5.5 0 0 1-1 0v-5h-5a.5.5 0 0 1 0-1h5v-5A.5.5 0 0 1 8 2Z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
